### ISSUE TITLE
USB dongle not detected on Ubuntu 20.04 with ARM64

### ISSUE THREAD

**Issue by hreuver0183** [bdbcat/oesenc_pi#94]
The issue:
The SG lock has been ported to Ubuntu 18.04 (bionic) which is essentially the same as RaspiOS Buster.
The oesenc_pi 4.2.12 can be downloaded from the pluginmanager and it works like it should.

For ARM64 on Ubuntu 20.04 however it does not work (focal/bullseye).

On the Pi4 with ubuntu 20.04:
`$ oeserverd-armhf -a`
`oeserverd Version 1.14`
`$ oeserverd-armhf -s`
`1`
`$ oeserverd-arm64 -a`
`oeserverd Version 1.17`
`$ oeserverd-arm64 -s`
`0`
`$ oeserverd-arm64 -t`
`0`
Included below: strace from the above oeserverd -s commands.
[oeserverd-problem.txt](https://github.com/bdbcat/oesenc_pi/files/4840186/oeserverd-problem.txt)

**Comment by hreuver0183** [comment]
Extra information.
On Ubuntu 18.04:
'odroidn2:~$ .local/bin/oeserverd -a'
'oeserverd Version 1.17'
'odroidn2:~$  .local/bin/oeserverd -s'
'1'
'odroidn2:~$ ldd .local/bin/oeserverd'
'.local/bin/oeserverd:'
'        linux-vdso.so.1 (0x0000007fab470000)'
'        libpthread.so.0 => /lib/aarch64-linux-gnu/libpthread.so.0 (0x0000007fab3d0000)'
'        libdl.so.2 => /lib/aarch64-linux-gnu/libdl.so.2 (0x0000007fab3bb000)'
'        libstdc++.so.6 => /usr/lib/aarch64-linux-gnu/libstdc++.so.6 (0x0000007fab227000)'
'        libgcc_s.so.1 => /lib/aarch64-linux-gnu/libgcc_s.so.1 (0x0000007fab203000)'
'        libc.so.6 => /lib/aarch64-linux-gnu/libc.so.6 (0x0000007fab0aa000)'
'        /lib/ld-linux-aarch64.so.1 (0x0000007fab445000)'
'        libm.so.6 => /lib/aarch64-linux-gnu/libm.so.6 (0x0000007faaff0000)'

On Ubuntu 20.04:
'odroidc4:~$ /usr/bin/oeserverd-arm64 -a'
'oeserverd Version 1.17'

'odroidc4:~$ /usr/bin/oeserverd-arm64 -s'
'0'
'odroidc4:~$ ldd /usr/bin/oeserverd-arm64'
'/usr/bin/oeserverd-arm64:'
'        linux-vdso.so.1 (0x0000ffff9b132000)'
'        /usr/lib/libgbm-compat.so (0x0000ffff9b0ca000)'
'        /usr/lib/aarch64-linux-gnu/libmali.so (0x0000ffff99efe000)'
'        libpthread.so.0 => /lib/aarch64-linux-gnu/libpthread.so.0 (0x0000ffff99eb9000)'
'        libdl.so.2 => /lib/aarch64-linux-gnu/libdl.so.2 (0x0000ffff99ea5000)'
'        libstdc++.so.6 => /lib/aarch64-linux-gnu/libstdc++.so.6 (0x0000ffff99cc3000)'
'        libgcc_s.so.1 => /lib/aarch64-linux-gnu/libgcc_s.so.1 (0x0000ffff99c9e000)'
'        libc.so.6 => /lib/aarch64-linux-gnu/libc.so.6 (0x0000ffff99b2d000)'
'        /lib/ld-linux-aarch64.so.1 (0x0000ffff9b102000)'
'        libgbm.so.1 => /lib/aarch64-linux-gnu/libgbm.so.1 (0x0000ffff99b0e000)'
'        libwayland-server.so.0 => /lib/aarch64-linux-gnu/libwayland-server.so.0 (0x0000ffff99aea000)'
'        libdrm.so.2 => /lib/aarch64-linux-gnu/libdrm.so.2 (0x0000ffff99ac7000)'
'        libwayland-client.so.0 => /lib/aarch64-linux-gnu/libwayland-client.so.0 (0x0000ffff99aa8000)'
'        libm.so.6 => /lib/aarch64-linux-gnu/libm.so.6 (0x0000ffff999fd000)'
'        libexpat.so.1 => /lib/aarch64-linux-gnu/libexpat.so.1 (0x0000ffff999c6000)'
'        libffi.so.7 => /lib/aarch64-linux-gnu/libffi.so.7 (0x0000ffff999ad000)'

Since both github and pluginmanagers use binary versions of oeserverd and libsglock I don't think the problem lies in my github compile.

**Comment by bdbcat** [comment]
First point:  Where did you get Ubuntu 20.04 (focal) for arm64?

**Comment by hreuver0183** [comment]
2 sources.
I prefer using the official RPI4 Ubuntu 20.04 as reference.
(Other is some non-official odroid port).

**Comment by hreuver0183** [comment]
`opencpn@raspi4-4-64:~$ /usr/bin/oeserverd-arm64 -a
oeserverd Version 1.17

opencpn@raspi4-4-64:~$ /usr/bin/oeserverd-arm64 -s
0
opencpn@raspi4-4-64:~$ ldd /usr/bin/oeserverd-arm64 
        linux-vdso.so.1 (0x0000ffffb082d000)
        libpthread.so.0 => /lib/aarch64-linux-gnu/libpthread.so.0 (0x0000ffffb0790000)
        libdl.so.2 => /lib/aarch64-linux-gnu/libdl.so.2 (0x0000ffffb077c000)
        libstdc++.so.6 => /lib/aarch64-linux-gnu/libstdc++.so.6 (0x0000ffffb059a000)
        libgcc_s.so.1 => /lib/aarch64-linux-gnu/libgcc_s.so.1 (0x0000ffffb0575000)
        libc.so.6 => /lib/aarch64-linux-gnu/libc.so.6 (0x0000ffffb0404000)
        /lib/ld-linux-aarch64.so.1 (0x0000ffffb07fd000)
        libm.so.6 => /lib/aarch64-linux-gnu/libm.so.6 (0x0000ffffb0359000)
`

**Comment by hreuver0183** [comment]
I tried to make it readable since the opload of the scripted output did not work.
Hope it helps somehow.

**Comment by bdbcat** [comment]
@hreuver0183 ...
I really want to find out if this is an experiment, or is this a viable beginning user configuration.  Makes a difference how I prioritize the investigation.

First:
So, I guess you are using the rPI Imager to download the Ubuntu 20.04?

Second point:
Where does oesenc come from? We do not build  managed plugin for arm64/20.04.  Are you building locally?

**Comment by hreuver0183** [comment]
Ubuntu 20.04 Focal and Bullseye are not ready for navigation yet. You see the same problems with GTK3 you see on x86_64 on arm64.

In my opinion early adapters should stick to Buster/arm64 or Bionic/arm64 for 2020 with the 5.2 release of OpenCPN. The release of OpenCPN 5.2 should be first priority.

I think the USB key dongle for ubuntu 20.04/arm64 should be made to work after, say, oktober-november  2020.

I think if the Ubuntu 20.04 version of OpenCPN should be ready at the beginning of next season it would be nice (approx. april 2021), since Ubuntu 20.04 will be the first system a lot of arm64 clones are ported to. It is the only official platform where Ubuntu supports the RPI4!

As for my RPI4, it is just a reference testing system. My navigation computer is the Odroid N2 (Ubuntu 18.04 this year, perhaps Ubuntu 20.04 next).

**Comment by hreuver0183** [comment]
After upgrading to oesenc_4.2.14 and reinstalling libusb-0.1 and libusb-1.0 and upgrading to OpenCPN-5.2.0 it worked.
(Libusb has been mentioned in the 5.1.xxx beta threat in CF).
Since the libsglarm64 and the oeserverd have not changed since june I must have forgotten one step somewhere.

I thought I had tested the situation thoroughly. My apologies.

Tested on : 
- Odroid N2 (Armbian Focal 20.04) opencpn-bionic withPM oesenc
- RPI4 (Official Ubuntu Focal 20.04) opencpn-focal with github oesenc

With Armbian the libwxgtk3.0-gtk3-0v5 is not available but the libwxgtk3.0-gtk3-0v5 is. Looks like it is really something in between bionic and focal. I used the opencpn bionic version there.
